crypto/compactcert/builder.go

```diff
@@ -122,7 +132,7 @@ func (b *Builder) Add(pos uint64, sig merklesignature.Signature) {
 
```

Comment:
In the function `Add` above:
The function is assuming that the caller of `Add` has already checked that the `sig` is not in the `Builder`. 
But this is not a valid assumption, since `Add` is an exported function, and the caller may call `Add` multiple times, without knowing that this is not allowed. When this happens, `b.signedWeight` will go out of sync with the real weight. 

Check for `Present` inside `Add` and report an error or just ignore the call if the sig is already in the builder.